docs/FRSKY_R9.md

```diff
@@ -70,31 +88,94 @@ The wiring is similar to the ExpressLRS docs found in the link below. Different
+
+2. Install the R9M module in your radio.  Enter the System Menu and navigate using the page buttons to the SD card page.
+
+3. Scroll to the FIRMWARE folder and select the r9m\_elrs\_bl.frk file.  Choose "Flash external module".
 
-In these docs it is suggested to download and use the "ST-LINK Utility" software. This software is pretty outdated (NRND = not recommended for new designs), and the new recommended tool is "STM32CubeProgrammer". However, there might be catches:
-- ST-LINK Utility appears to not work with newer ST-Link programmers.
-- STM32CubeProgrammer/STM32CubeIDE is quite nasty with which ST-Link programmer is used. It seems ST really wants their tools to only work well with "original" or legit ST-Link programmers. I especially had significant issues with getting the cheap and widely available 8$ STLinkV2 usb-stick-like clones to work with STM32CubeProgrammer/STM32CubeIDE, and they then never worked reliably or to my satisfaction.
-- It seems it can happen that ST-LINK Utility and STM32CubeProgrammer/STM32CubeIDE do not like each other.
-- STM32CubeProgrammer runs per default at a high SWD frequency, like 4000 kHz, on the 8$ STLinkV2 clones. Setting the frequency manually to 480 kHz and using short wires to the device makes it work more reliable.
-- STM32CubeProgrammer will likely require new firmware be installed on the ST-Link V2 or compatible device via the "Firmware upgrade" button" the first time it is used.
+4. Follow the instructions in the next section to flash mLRS.
 
-So, if you use or want to use one of these 8$ STLinkV2 usb-stick-like clones and you have trouble with STM32Cube, you probably want to install ST-LINK Utility and see how that works for you. You will then (likely) not be able to flash from STM32CubeIDE, but have to go with the ST-LINK Utility.
+If you have already flashed via ST-Link, you can continue to flash updates via ST-Link the way you always have, or you can follow these steps to switch to flashing from the radio.  To do this, you will flash the ELRS bootloader by using ST-Link one last time:
 
-If anyone has deeper/better insight into the STM32CubeProgrammer/STM32CubeIDE vs ST-LINK Utility and cheap STLinkV2 programmer clones issues, please help :)
+1. Download the [r9m\_bootloader.bin](https://github.com/ExpressLRS/ExpressLRS/blob/master/src/bootloader/r9m_bootloader.bin?raw=true) file from the ExpressLRS git repository.
+2. Flash it to the beginning of the flash (0x8000000) using ST-Link and STM32CubeProgrammer.
 
-If you wonder what STLink programmer I (olliw42) am using: Every NUCLEO board comes also with a STLink programmer... and these board are relatively cheap too, about 15Eur.
+3. Follow the instructions below to flash mLRS.
+
+#### Flash/Update the mLRS firmware ####
+
+1. Build using run\_make\_firmwares3.py or download the latest mLRS firmware.
```

Comment:
I for sure would not mention the run_make_xxx script at all. The purpose is to make it easy for the uisers, isn't it. So I would just say something like "Download the latest mLRS firmware from the github repository. Choose the file with a "elrs-bl" in the name and the ".elrs" extension".
Dito below.

Reply:
I agree about the purpose, but I want to provide this hint for anyone building from  source.  Last time I checked, this script wasn't mentioned in the build instructions.

Reply:
again, this script is not for regular usage by anyone ... if it continues to cause confusion I will remove it
I think you will just trip off most readers ... but you guys decide, I just share my thoughts :)

Reply:
you always can enter global build flags in the IDE, or create targets with specific global build flags, etc. No one said it's easier. And I repeat, this py script is just not aiming at being used by everyone.
What I just can't understand is your mindset here. IMHO, ideally, users should not have to compile any firmware. And as a matter of fact, they don't have to.
Anyway, you guys pl decide :)

Reply:
For now, I'll remove the references to the script, but mention the define which is required.  But I think info on building these special images needs to be documented somewhere, perhaps in the mLRS README.md.